lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/HeartBeatManager.cs

```diff
@@ -90,6 +93,24 @@ internal sealed class HeartBeatManager : IHeartBeatManager
                 _clock = settings.RuntimeClock;
                 _heartBeatPeriodInMillSeconds = settings.HeartBeatPeriodInMs;
                 _maxHeartbeatRetries = settings.MaxHeartbeatRetries;
+                _driverConnection = new Lazy<IDriverConnection>(
```

Comment:
This looks a lot like a nicer approach to `InjectionFuture`? Or should we use `InjectionFuture` here?